include/spdlog/details/registry_shared.h

```diff
@@ -0,0 +1,66 @@
+/*************************************************************************/
+/* spdlog - an extremely fast and easy to use c++11 logging library.     */
+/* Copyright (c) 2014 Gabi Melman.                                       */
+/*                                                                       */
+/* Permission is hereby granted, free of charge, to any person obtaining */
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#pragma once
+
+#include "registry.h"
+
+namespace spdlog
+{
+namespace details
+{
+template <class Mutex> class registry_shared_t : public registry_t<Mutex>
+{
+public:
+    static std::shared_ptr<registry_shared_t<Mutex> > instance()
+    {
+        if(!_instance)
+            _instance = std::make_shared<registry_shared_t<Mutex> >();
+        return _instance;
+    }
+
+    static void set_registry(std::shared_ptr<registry_shared_t<Mutex> > registry)
+    {
+        _instance = registry;
+    }
+
+    registry_shared_t<Mutex>() : registry_t<Mutex>() {}
+
+private:
+    static std::shared_ptr<registry_shared_t<Mutex> > _instance;
+};
+
+#ifdef SPDLOG_USE_SHARED_REGISTRY
+#   ifdef SPDLOG_NO_REGISTRY_MUTEX
+        typedef registry_shared_t<spdlog::details::null_mutex> registry;
+#   else
+        typedef registry_shared_t<std::mutex> registry;
+#   endif //SPDLOG_NO_REGISTRY_MUTEX
+#endif //SPDLOG_USE_SHARED_REGISTRY
+
+template<class Mutex>
+std::shared_ptr<registry_shared_t<Mutex> > registry_shared_t<Mutex>::_instance = nullptr;
+
+}
+}
```

Comment:
If the only thing that is added is the  set_registry function, Why not just to add this function to the orig registry class . This way instead of all this new code and ifdefs, only 3 lines of code would be added.